numalogic/registry/mlflow_registry.py

```diff
@@ -139,20 +142,22 @@ def load(
         version: str = None,
     ) -> Optional[ArtifactData]:
         model_key = self.construct_key(skeys, dkeys)
-        try:
-            if (latest and version) or (not latest and not version):
-                raise ValueError("Either One of 'latest' or 'version' needed in load method call")
 
-            elif latest:
+        if (latest and version) or (not latest and not version):
+            raise ValueError("Either One of 'latest' or 'version' needed in load method call")
+
+        try:
+            if latest:
+                cached_artifact = self._load_from_cache(model_key)
```

Comment:
are we implementing cache only if its latest version?

Reply:
what do we do when user wants to switch to an older version?

Reply:
Yes only for latest. So the thought here is that user only cares about caching in the streaming ML pipeline. Which will always get the latest version. If not, user needs to delete the current version, and point the latest to the older version. Then caching will work again.
WDYT? Otherwise it can be complicated with versioning, since the key will clash.

Reply:
yeah I think the model switching experience has to be improved for users, lets keep this in mind while designing the redis model storage.

Reply:
@s0nicboOm FYI ^